src/main/java/moody/parser/Parser.java

```diff
@@ -0,0 +1,197 @@
+package moody.parser;
+
+import moody.exceptions.InvalidCommandException;
+import moody.exceptions.TaskInputException;
+import moody.commands.*;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user input into Command objects.
+ * The Parser class is responsible for interpreting the commands provided by the user
+ * and returning the corresponding Command objects.
+ */
+public class Parser {
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /**
+     * Parses the given command string into a Command object.
+     * The method splits the command string into the command keyword and arguments,
+     * then creates and returns the appropriate Command based on the keyword.
+     *
+     * @param input The command string to parse.
+     * @return The Command object corresponding to the parsed command.
+     * @throws InvalidCommandException If the command is not recognized.
+     * @throws TaskInputException If there is an issue with the task input.
+     */
+    public static Command parse(String input) throws InvalidCommandException, TaskInputException {
+        String[] parts = input.split(" ", 2); // Split into command and arguments
+
+        String command = parts[0].toLowerCase(); // Get the command keyword
```

Comment:
To avoid confusion with Command class, you might want to change the variable name to 'commandWord' instead